scripts/src/java/org/oppia/android/scripts/kdoc/KDocCheck.kt

```diff
@@ -0,0 +1,284 @@
+ * exempted for the KDoc check.
+ *
+ * Usage:
+ *   bazel run //scripts:kdoc_check -- <path_to_directory_root>
+ *
+ * Arguments:
+ * - path_to_directory_root: directory path to the root of the Oppia Android repository.
+ *
+ * Example:
+ *   bazel run //scripts:kdoc_check -- $(pwd)
+ */
+fun main(vararg args: String) {
+  // Path of the repo to be analyzed.
+  val repoPath = "${args[0]}/"
+
+  val kdocExemptiontextProto = "scripts/assets/kdoc_exemptions"
+
+  // List of annotation entries which when present on an element, the element does not needs to be
+  // checked for a KDoc.
+  val kDocNotRequiredAnnotationEntryList = listOf<String>(
+    "Rule",
+    "Mock",
+    "Test",
+    "Before",
+    "Captor",
+    "BeforeClass",
+    "After",
+    "AfterClass",
+    "Inject",
+    "Provides"
+  )
+
+  // A list of all the files to be exempted for this check.
+  val kdocExemptionList = loadKdocExemptionsProto(kdocExemptiontextProto).getExemptedFilePathList()
+
+  // A list of all kotlin files in the repo to be analyzed.
+  val searchFiles = RepositoryFile.collectSearchFiles(
+    repoPath = repoPath,
+    expectedExtension = ".kt",
+    exemptionsList = kdocExemptionList
+  )
+
+  // A list of all the files missing KDoc(s).
+  val matchedFiles = searchFiles.filter { file ->
+    hasKdocFailure(file, kDocNotRequiredAnnotationEntryList)
+  }
+
+  if (matchedFiles.isNotEmpty()) {
+    throw Exception("KDOC CHECK FAILED")
+  } else {
+    println("KDOC CHECK PASSED")
+  }
+}
+
+/** IntelliJ Project object, which is required to generate a [KtFile] from a Kotlin file. */
+private val project by lazy {
+  val config = CompilerConfiguration()
+  config.put(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, MessageCollector.NONE)
+  KotlinCoreEnvironment.createForProduction(
+    Disposer.newDisposable(),
+    config,
+    EnvironmentConfigFiles.JVM_CONFIG_FILES
+  ).project
+}
+
+/**
+ * Genrates a [KtFile] from a Kotlin file.
+ *
+ * @param codeString the string equivalent of the Kotlin file
+ * @param fileName name of the file
+ * @return the generated [KtFile] for the given file
+ */
+private fun createKtFile(codeString: String, fileName: String): KtFile {
+  return PsiManager.getInstance(project)
+    .findFile(LightVirtualFile(fileName, KotlinFileType.INSTANCE, codeString)) as KtFile
+}
+
+/**
+ * Checks whether a file has a KDoc presence failure.
+ *
+ * @param file the file to be checked
+ * @param kDocNotRequiredAnnotationEntryList the list of annotation entries which when present
+ *     on an element, the element does not needs to be checked for a KDoc.
+ * @return whether the file has a KDoc presence failure
+ */
+private fun hasKdocFailure(file: File, kDocNotRequiredAnnotationEntryList: List<String>): Boolean {
+  val ktFile = createKtFile(file.asString(), file.name)
+  return ktFile.children.fold(initial = false) { isFailing, elem ->
+    val childFailingKdoc = if (elem is KtElement) elementIterator(
```

Comment:
This if-check should have curly braces. We only omit curly braces on statements if they fit entirely on one line.

Reply:
Done.